### ISSUE TITLE
Scene Graphs

### ISSUE THREAD

**Issue by define-private-public** [zacharycarter/zengine#23]
Many game engines have these.  What are we going to do for ours?  I don't think we need to work on this right now, but we should cook up some ideas:

1. Should not be forced on the user to make a game.  It's completely fine to make a game without one, though recommended.  I really don't like engines that force scene graphs onto the programmer.
2. Engines like Urho have serialization/deserialization for their scene graphs.  Might be nice to have for ours.
3. I think this is a plus (or kinda necessary) if we want to have a level editing tool.

**Comment by zacharycarter** [comment]
I agree we should eventually add one. I haven't looked at Urho3d's but I have looked at Horde3d's and I like theirs. Maybe we can steal ideas from both. Either way I agree with all three of your points.
